scim-server/src/main/java/org/apache/directory/scim/server/rest/BaseResourceTypeResourceImpl.java

```diff
@@ -382,35 +358,51 @@ private ScimException notFoundException(String id) {
     return new ScimException(Status.NOT_FOUND, "Resource " + id + " not found");
   }
 
-  private void validatePreconditions(String id, EntityTag etag) {
-    ResponseBuilder response = request.evaluatePreconditions(etag);
-    if (response != null) {
-      throw new WebApplicationException(response
-        .entity(new ErrorResponse(Status.PRECONDITION_FAILED, "Failed to update record, backing record has changed - " + id))
-        .build());
-    }
-  }
+//  private void validatePreconditions(String id, EntityTag etag) {
+//    ResponseBuilder response = request.evaluatePreconditions(etag);
+//    if (response != null) {
+//      throw new WebApplicationException(response
+//        .entity(new ErrorResponse(Status.PRECONDITION_FAILED, "Failed to update record, backing record has changed - " + id))
+//        .build());
+//    }
+//  }
+//
+//  private EntityTag requireEtag(ScimResource resource) throws ScimException {
+//    try {
+//      return etagGenerator.generateEtag(resource);
+//    } catch (EtagGenerationException e) {
+//      throw new ScimException(Status.INTERNAL_SERVER_ERROR, "Failed to generate the etag", e);
+//    }
+//  }
+//
+//  private EntityTag etag(ScimResource resource) {
+//    try {
+//      return etagGenerator.generateEtag(resource);
+//    } catch (EtagGenerationException e) {
+//      log.warn("Failed to generate etag for resource", e);
+//      return null;
+//    }
+//  }
```

Comment:
Still getting my footing in reviewing this PR. What's the reason for having these lines commented-out at the moment?

Reply:
Good catch, i though I deleted all of the dead code, I'll take another pass through this PR!

Reply:
Removed